### PR TITLE
Add type annotations, update documentation

### LINKED ISSUE #74: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_bus_device/spi_device.py:55
- [ ] adafruit_bus_device/spi_device.py:86
- [ ] adafruit_bus_device/i2c_device.py:44
- [ ] adafruit_bus_device/i2c_device.py:52
- [ ] adafruit_bus_device/i2c_device.py:69
- [ ] adafruit_bus_device/i2c_device.py:87
- [ ] adafruit_bus_device/i2c_device.py:141

### PR DESCRIPTION
**Pull request by tekktrik**
Resolves #74, and also adds additional documentation in the docstrings about argument types.

### CODE REVIEW

File: adafruit_bus_device/i2c_device.py [hunk 1]
```diff
@@ -41,15 +48,15 @@ class I2CDevice:
...
+    def __init__(self, i2c: I2C, device_address: int, probe: bool = True):
 
         self.i2c = i2c
         self.device_address = device_address
 
         if probe:
             self.__probe_for_device()
 
-    def readinto(self, buf, *, start=0, end=None):
+    def readinto(self, buf: bytearray, *, start: int = 0, end: Optional[int] = None):
```
Comment (tannewt) [inline adafruit_bus_device/i2c_device.py]: Don't we have a `WritableBuffer` type? I think it should work for more than just bytearray.

File: adafruit_bus_device/i2c_device.py [hunk 2]
```diff
@@ -66,7 +73,7 @@ def readinto(self, buf, *, start=0, end=None):
             end = len(buf)
         self.i2c.readfrom_into(self.device_address, buf, start=start, end=end)
 
-    def write(self, buf, *, start=0, end=None):
+    def write(self, buf: bytearray, *, start: int = 0, end: Optional[int] = None):
```
Comment (tannewt) [inline adafruit_bus_device/i2c_device.py]: This should use `ReadableBuffer` I think.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question about the buffer types. Thanks for doing this!

File: adafruit_bus_device/i2c_device.py (hunk 2 above)
Comment (tekktrik) [reply adafruit_bus_device/i2c_device.py]: It should, I think I based it off the docstring, so I'll change the type annotation and update the docstring as well.

File: adafruit_bus_device/i2c_device.py (hunk 1 above)
Comment (tekktrik) [reply adafruit_bus_device/i2c_device.py]: See comment below!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**